clouds/aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,745 @@
+
+
+Usage:
+1. Import this module to create AWS network resources and Security Group with Inbound and Outbound rules :
+
+    from create_aws_network_resources import create_network_resources, create_security_group, ec2_add_security_rules
+
+    resources = create_aws_network_resources(cidr_block="10,0,0,0/16", site_name="demo", subnet_count=3)
+    security_grp_id = create_security_group(vpc_id=resources['vpc_id'], sec_grp_name="security-demo")
+    ec2_add_security_rules(security_grp_id=security_grp_id, rules_json_file="security_rules_example.json")
+
+2.  python create_aws_network_resources.py
+        --aws_access_key_id=<access-key> --aws_secret_access_key=<secret> --region <aws_region>
+"""
+
+import json
+import os
+import sys
+import time
+from ipaddress import AddressValueError, IPv4Network
+
+import boto3
+import click
+from aws_utils import verify_aws_config, verify_aws_credentials
+from botocore.exceptions import ClientError
+from simple_logger.logger import get_logger
+
+LOGGER = get_logger(name=__name__)
+
+
+def ec2_assign_name_tag(ec2_client, res_id, tag_name) -> None:
+    """
+    Assign 'Name' tag value to EC2 resource
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         res_id (str): The ID of resource(which can be of 'vpc', 'subnet', etc)
+         tag_name (str): The value of 'Name' tag
+    """
+    ec2_client.create_tags(
+        Resources=[
+            res_id,
+        ],
+        Tags=[
+            {
+                "Key": "Name",
+                "Value": tag_name,
+            },
+        ],
+    )
+
+
+def get_subnets_cidr(cidr_block, subnet_count) -> list:
+    """
+    Calculates Subnet CIDR address blocks within given VPC CIDR block
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         cidr_block (str): The IPv4 network of VPC
+         subnet_count (str): The number of additional bits with which to extend the prefix
+
+    Returns:
+         list: The list of subnet CIDR address blocks
+
+    Raises:
+         AddressValueError: If cidr_block provided in invalid for IPv4 address
+    """
+    subnet_range = []
+    try:
+        net = IPv4Network(address=cidr_block)
+        prefix = net.prefixlen + (subnet_count * 2)
+        for subnet in net.subnets(new_prefix=prefix):
+            subnet_range.append(str(subnet))
+        return subnet_range
+    except AddressValueError:
+        LOGGER.error(f"Address is invalid for IPv4: {cidr_block}")
+        raise
+
+
+def get_availability_zones(ec2_client) -> list:
+    """
+    Provides a list of zone IDs in availability zone of current AWS Region
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+
+    Returns:
+         list: The list of Zone IDs
+    """
+    zones = []
+    av_zones = ec2_client.describe_availability_zones()
+    for zone in av_zones.get("AvailabilityZones"):
+        zones.append(zone.get("ZoneId"))
+    LOGGER.info(f"Availability zones : { zones }")
```

Comment:
code which uses this function assumes that something is returned; do not use `get` - get by key and fail if key does not exist

Reply:
and raise if zones is empty before return

Reply:
done